sphinx/tutorial/cylc/furthertopics/cylc-broadcast.rst

```diff
@@ -0,0 +1,132 @@
+   cd ~/cylc-run/tutorial-broadcast
+
+Copy the following configuration into a ``suite.rc`` file:
+
+.. code-block:: cylc
+
+   [scheduling]
+       initial cycle point = 1012
+       [[dependencies]]
+           [[[R1]]]
+               graph = wipe_log => announce
+           [[[PT1H]]]
+               graph = announce[-PT1H] => announce
+
+   [runtime]
+       [[wipe_log]]
+           # Delete any files in the suite's "share" directory.
+           script = rm "${CYLC_SUITE_SHARE_DIR}/*"
+
+       [[announce]]
+           script = echo "${CYLC_TASK_CYCLE_POINT} - ${MESSAGE}" >> "${FILE}"
+           [[[environment]]]
+               WORD = ni
+               MESSAGE = We are the knights who say \"${WORD}\"!
+               FILE = "${CYLC_SUITE_SHARE_DIR}/knights"
+
+We now have a suite with an ``announce`` task which runs every hour, writing a
+message to a log file (``share/knights``) when it does so. For the first cycle
+the log entry will look like this::
+
+   10120101T0000Z - We are the knights who say "ni"!
+
+The ``cylc broadcast`` task enables us to change runtime configuration whilst
+the suite is running. For instance we could change the value of the ``WORD``
+environment variable using the command::
+
+   cylc broadcast tutorial-broadcast -n announce -s "[environment]WORD=if"
+
+* ``tutorial-broadcast`` is the name of the suite.
+* ``-n announce`` tells cylc we want to change the runtime configuration of the
+  ``announce`` task.
+* ``-s "[environment]WORD=if"`` changes the value of the ``WORD`` environment
+  variable to ``it``.
+
+Run the suite then try using the ``cylc broadcast`` command to change the
+message::
+
+   cylc run tutorial-broadcast
+   cylc broadcast tutorial-broadcast -n announce -s "[environment]WORD=it"
+   ...
+
+Inspect the ``share/knights`` file, you should see the message change at
+certain points.
+
+Stop the suite::
+
+   cylc stop tutorial-broadcast
+
+
+In-Situ Example
+---------------
+
+We can call the ``cylc broadcast`` command from within a task's script.
+
+Add the following recurrence to the ``dependencies`` section:
+
+.. code-block:: cylc
+
+           [[[PT3H]]]
+               graph = announce[-PT1H] => change_word => announce
+
+The ``change_word`` task runs the ``cylc broadcast`` command to randomly
+change the ``WORD`` environment variable used by the ``announce`` task.
+
+Add the following runtime configuration to the ``runtime`` section:
+
+.. code-block:: cylc
+
+       [[change_word]]
+           script = """
+               # Select random word.
+               IFS=',' read -r -a WORDS <<< $WORDS
+               WORD=${WORDS[$(date +%s) % ${#WORDS[@]}]}
+
+               # Broadcast random word to the announce task.
+               cylc broadcast $CYLC_SUITE_NAME -n announce -s "[environment]WORD=${WORD}"
+           """
+           [[[environment]]]
+               WORDS = ni, it, eche eche foo vatang
+
+Run the suite and inspect the log, you should see the message change randomly
```

Comment:
'...log. You should...' Forgive me for being pedantic but this type of sentence structure does not flow well / does not seem very grammatically sound and hence I have requested a split to two sentences in the main body of the tutorial in all similar cases, of which there are several. Ideally we can change equivalent structures such as this one in the further topics as well. Bad text cohesion distracts from the content.